botocove/cove_host_account.py

```diff
@@ -49,18 +40,40 @@ def __init__(
 
         self.thread_workers = thread_workers
 
-        self.sts_client = self._get_boto3_sts_client(assuming_session)
-        self.org_client = self._get_boto3_org_client(assuming_session)
+        # I would pass a session around and instantiate the necessary client directly.
+        # I think we don't need special threading settings in these clients any more.
+        # CoveHostAccount is set up in a single thread.
+        # Is the client logging output considered part of the public interface?
```

Comment:
Is CoveHostAccount part of the module's public interface?

I'd like the answer to be "no" so that we have more freedom to rearrange and simplify this as we wish. :-)

I'm guessing these custom boto3 client setups were required when botocove was run in two phases, but it doesn't seem necessary any more.

Reply:
That's good to know! It means we could now really simplify that part of the code. But from the other comments around here I've come to the conclusion that this refactoring mixed with a big change in the functionality is too much change for one PR!